### PR TITLE
A couple extra features

### PR DESCRIPTION
**Pull request by jadwahab**

### CODE REVIEW

File: examples/create_tx/create_tx.go [hunk 1]
```diff
@@ -20,9 +20,9 @@ func main() {
...
 	wif, _ := bsvutil.DecodeWIF("KznvCNc6Yf4iztSThoMH6oHWzH9EgjfodKxmeuUGPq5DEX5maspS")
 
-	err := tx.SignAuto(&bt.InternalSigner{PrivateKey: wif.PrivKey, SigHashFlag: 0})
-	if err != nil {
-		log.Fatalln(err.Error())
+	inputsSigned, err := tx.SignAuto(&bt.InternalSigner{PrivateKey: wif.PrivKey, SigHashFlag: 0})
+	if err != nil && len(inputsSigned) > 0 {
+		log.Fatal(err.Error())
 	}
 	log.Println("tx: ", tx.ToString())
```
Comment (mrz1836) [inline examples/create_tx/create_tx.go:27]: You want to leave these logs in there?

File: examples/create_tx_with_opreturn/create_tx_with_opreturn.go [hunk 2]
```diff
@@ -27,8 +27,8 @@ func main() {
 
 	wif, _ := bsvutil.DecodeWIF("L3VJH2hcRGYYG6YrbWGmsxQC1zyYixA82YjgEyrEUWDs4ALgk8Vu")
 
-	err = tx.SignAuto(&bt.InternalSigner{PrivateKey: wif.PrivKey, SigHashFlag: 0})
-	if err != nil {
+	inputsSigned, err := tx.SignAuto(&bt.InternalSigner{PrivateKey: wif.PrivKey, SigHashFlag: 0})
+	if err != nil && len(inputsSigned) > 0 {
 		log.Fatal(err.Error())
 	}
 	log.Println("tx: ", tx.ToString())
```
Comment (mrz1836) [inline examples/create_tx_with_opreturn/create_tx_with_opreturn.go:34]: Same - logs here as well

**Review by mrz1836 — COMMENTED** [review]
@jadwahab minor comments about logs

**Review by mrz1836 — APPROVED** [review]
LGTM - added some tests and made a few minor changes @jadwahab

**Comment by mrz1836** [comment]
Moved away from Travis into Github Actions for linting and running tests @jadwahab

File: internalsigner.go [hunk 3]
```diff
@@ -23,14 +23,13 @@ func (is *InternalSigner) Sign(index uint32, unsignedTx *Tx) (signedTx *Tx, err
 		is.SigHashFlag = sighash.AllForkID
 	}
 
-	sh, err := unsignedTx.GetInputSignatureHash(index, is.SigHashFlag)
-	if err != nil {
+	var sh []byte
+	if sh, err = unsignedTx.GetInputSignatureHash(index, is.SigHashFlag); err != nil {
```
Comment (jadwahab) [inline internalsigner.go:27]: @mrz1836 does your IDE automatically do this?

Comment (mrz1836) [reply internalsigner.go:27]: Yeah my IDE complains, [shadow var](https://yourbasic.org/golang/gotcha-shadowing-variables/) again... overwriting `err`

**Pull request merged**